pkg/config/v1alpha1/types.go

```diff
@@ -135,6 +135,10 @@ type ControllerHealth struct {
 	// LivenessEndpointName, defaults to "healthz"
 	// +optional
 	LivenessEndpointName string `json:"livenessEndpointName,omitempty"`
+
+	// StartupEndpointName, defaults to "startz"
+	// +optional
+	StartupEndpointName string `json:"startupEndpointName,omitempty"`
```

Comment:
I know this is where the other endpoint names live but this is deprecated.  Do we want to keep it here with the risk of it being moved and/or removed for configuration?

Reply:
I was not aware of this file being deprecated. Where should it be instead? I didn't find another place where the endpoint name is defined. I suppose that the risk for StartupEndpointName is the same than for the others, is it correct?

Reply:
>// ControllerHealth defines the health configs.
//
// Deprecated: The component config package has been deprecated and will be removed in a future release. Users should migrate to their own config implementation, please share feedback in https://github.com/kubernetes-sigs/controller-runtime/issues/895.

There is this PR that removes this file https://github.com/kubernetes-sigs/controller-runtime/pull/2648 because most of the config is deprecated.  Adding this functionality there doesn't seem like it will stay long as the plan would be to get rid of this configuration. 

Now that doesn't mean it doesn't have a place defined in this repo, but I would probably place it as an option and not as deprecated option within the

Reply:
Understood, thank you. However, from my point of view, isn't it better to remove this config from my change when https://github.com/kubernetes-sigs/controller-runtime/pull/2648 is merged? Because right now it's still in progress

After merging your change, I can rebase my change to bring your changes and then I can remove my code from the deprecated files.

Reply:
Yours may get merged first, so with that if the maintainers lgtm/approve, I can work on finding a spot for this to move it out of the deprecated struct.